rfcs/0211-close-transactions-on-daily-closing.md

```diff
@@ -0,0 +1,100 @@
+- Feature Name: `close_transactions_on_daily_closing`
+- Start Date: 2023-10-18
+- RFC PR: [fiskaltrust/middleware#211](https://github.com/fiskaltrust/middleware/pull/211)
+- Tracking Issue: [fiskaltrust/middleware#212](https://github.com/fiskaltrust/middleware/issues/212)
+- Markets: `DE`
+
+# Summary
+
+We want to introduce a ftReceiptCaseFlag to the Daily-Closing that closes all open transactions on the TSE.
+
+# Motivation
+
+Open transactions on the TSE can lead to lots of problems and it already is our recommendation to close open transactions before performing the Daily-Closing.
+Right now this has to be done manually so this feature makes it more easy for PosCreators to do so.
+Originally we intended the Daily-Closing to have this behaviour per default.
+
+# Guide-level explanation
+
+If the the ftReceiptCaseFlag `0x0000000200000000` is set when sending a Daily-Closing all open transactions on the TSE will be closed prior to executing the daily closing.
+
+> ***Example:***
+> 
+> 1. A Zero-Receipt Request with the ftReceiptCase `0x4445000100800002` (Zero-Receipt with Request TSE-Info flag `0x800000`) is sent and in the Response we see there are three open transactions on the TSE.
+>    ```json
+>    {
+>      "CurrentStartedTransactionNumbers": [ 10, 12, 13 ]
+>    }
+>    ```
+> 2. A Daily-Closing Request with the ftReceiptCase `0x4445000300000007` (Daily-Closing with Close Open Transactions flag `0x200000000`) is sent.
+>    The open transactions are closed by the middleware.
+> 3. A Zero-Receipt Request with the ftReceiptCase `0x4445000100800002` is sent again in the Response there are no open transactions on the TSE.
+>    ```json
+>    {
+>      "CurrentStartedTransactionNumbers": []
+>    }
+>    ```
+
+# Reference-level explanation
+
+The Daily-Closing already gets the TSE-Info from the TSE and the middleware already has the ability to close open transactions on the TSE as done in the Fail-Transaction Receipt.
+
+In the we now introduce a check for the new flag and close the transactions if it is set.
+
+```cs
+if (request.HasCloseOpenTransactionsOnTseFlag() && tseInfo.CurrentStartedTransactionNumbers!.Any())
+{
+    await CloseOpenTransactionsOnTseAsync(tseInfo.CurrentStartedTransactionNumbers);
+}
+```
+
+The method `CloseOpenTransactionsOnTseAsync` looks like this which is inspired by the current [Fail-Transaction Receipt](https://github.com/fiskaltrust/middleware/blob/1a9abd80430e9dfecdd17289024e9d19e798d19b/queue/src/fiskaltrust.Middleware.Localization.QueueDE/RequestCommands/FailTransactionReceiptCommand.cs#L64-L77).
+
+```cs
+private CloseOpenTransactionsOnTseAsync(IEnumerable<long> currentStartedTransactionNumbers)
```

Comment:
This only closes the open transactions of the current Queue, right (since `transactionsToClose` is not used, but the data from the `_openTransactionsRepo`)?

Reply:
I've clarified the code a bit. It should finish the transactions on the TSE and remove them from the open transactions in the queue.